pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/realtimetoofflinesegments/RealtimeToOfflineSegmentsTaskExecutor.java

```diff
@@ -165,31 +176,89 @@ protected List<SegmentConversionResult> convert(PinotTaskConfig pinotTaskConfig,
 
     SegmentProcessorConfig segmentProcessorConfig = segmentProcessorConfigBuilder.build();
 
-    List<RecordReader> recordReaders = new ArrayList<>(numInputSegments);
+    // Check if this is an upsert table from the pre-computed config value
+    boolean isUpsertTable =
+        Boolean.parseBoolean(configs.getOrDefault(RealtimeToOfflineSegmentsTask.IS_UPSERT_TABLE, "false"));
```

Comment:
I feel we should derive this from the table config instead of putting a separated config?